hack/generator/pkg/astmodel/type_name.go

```diff
@@ -166,3 +166,19 @@ func (typeName TypeName) Plural() TypeName {
 
 	return MakeTypeName(typeName.PackageReference, flect.Pluralize(typeName.name))
 }
+
+// DebugDescription adds a description of the current type to the passed builder
+// builder receives the full description, including nested types
+// types is a dictionary for resolving named types
+func (typeName TypeName) WriteDebugDescription(builder *strings.Builder, types Types) {
+	builder.WriteString(typeName.PackageReference.String())
+	builder.WriteString("/")
+	builder.WriteString(typeName.name)
+	builder.WriteString(":")
+
+	if definition, ok := types[typeName]; ok {
```

Comment:
Might want to examine the TypeName to determine if it is/isn't a `LocalPackageReference` and then only follow the reference if it is? If it's something pointing to an external type then you can instead print something that clarifies that (and avoid the `NOTDEFINED`?)

Reply:
Good idea.

Reply:
Done.